### DISCUSSION TITLE
Kepware connection problem with Python - BadSessionNotActivated

### DISCUSSION THREAD

**Discussion by Steineklopfer** [FreeOpcUa/python-opcua#1441] [Q&A]
I've got a problem connecting to my Kepware Server...
I copied the example and shortened it a bit just to see if I'm able to get a connection going.

My server is configured to accept unsecure connections without a certificate from the localhost.
I also configured a Kepware simulator driver with a few tags.

```
#! python3

from opcua import Client
from time import sleep
#import win32com.client

client = Client("opc.tcp://localhost:49320")
#logging.basicConfig(level=logging.WARN)
client = Client("opc.tcp://localhost:49320/OPCUA/SimulationServer/")

try:
    client.connect()
    root = client.get_root_node()
    print("Root is", root)
    print("childs of root are: ", root.get_children())
    print("name of root is", root.get_browse_name())
    objects = client.get_objects_node()
    print("childs og objects are: ", objects.get_children())
except Exception as e:
    print(str(e))
else:
    print("Success")
    client.disconnect()
#finally:
```

tried both clients but that's not the problem it seems.
I get the error:

```
cryptography is not installed, use of crypto disabled
Requested session timeout to be 3600000ms, got 60000ms instead
ServiceFault from server received  in response to ActivateSessionRequest
"The session cannot be used because ActivateSession has not been called."(BadSessionNotActivated)
```

ok cryptography is not installed but so far I don't intend to use certificates since my application and the server will run on the same computer. So far we've been using c# with quickopc but we are trying to switch to python and a more universal approach.

Can anyone tell what might be the problem?

**Comment by AndreasHeine** [comment]
> cryptography is not installed, use of crypto disabled
> Requested session timeout to be 3600000ms, got 60000ms instead

is just logging stuff!

> ServiceFault from server received in response to ActivateSessionRequest
"The session cannot be used because ActivateSession has not been called."(BadSessionNotActivated)

thats the error! ServiceFault's comes typically from Server

what Versions your at (inkl. KEPServerEX Version?)

**Comment by Steineklopfer** [comment]
Yes, I thought so too :) 
I tried Kepware Server 6.7 and now the newest 6.11.
And I used pip to install opcua. It downloaded version 0.98.13 for me.

**Comment by Steineklopfer** [comment]
Ok. Probably I was too tired to see the deprecated message yesterday night. Now I tried the new Kepware example that comes with the asyncio v 0.9.92
The error remains but it reveals a bit more detail:

```
WARNING:asyncua.client.client:Requested session timeout to be 3600000ms, got 60000ms instead
WARNING:asyncua.client.ua_client.UASocketProtocol:ServiceFault (BadIdentityTokenInvalid, diagnostics: DiagnosticInfo(SymbolicId=None, NamespaceURI=None, Locale=None, LocalizedText=None, AdditionalInfo=None, InnerStatusCode=None, InnerDiagnosticInfo=None)) from server received  in response to ActivateSessionRequest
asyncua.ua.uaerrors._auto.BadIdentityTokenInvalid: "The user identity token is not valid."(BadIdentityTokenInvalid)
```

**Reply by AndreasHeine** [reply]
is the server password protected? or with usercertificates?

**Comment by Steineklopfer** [comment]
No it should not. I mean it has a server certificate that was generated during installation but I turned that off and specifically selected "none - insecure" in the UA configuration for localhost. I also never entered a password somewhere and left that empty.

**Comment by Steineklopfer** [comment]
Ok it seems there was another setting somewhere else where I had to check "allow anonymous login" so thanks for pointing me in the right direction! 
A bit confusing if you have different spots in the configuration you have to check for similar things...

**Reply by AndresParra11** [reply]
Hello, I would like to ask you where you marked "allow anonymous login", I have the same error. Thank you!

**Reply by Steineklopfer** [reply]
It is in "project properties"
Basically not in the OPC UA configuration of the server but in the properties of the project on the server there is another subsection which is called "OPC UA"  as well. There you set "allow anonymous login" to true.
